src/app/dim-ui/Sheet.tsx

```diff
@@ -112,110 +128,52 @@ export default function Sheet({
+  const [disabled, setParentDisabled] = useDisableParent(forceDisabled);
 
-  /** When the sheet stops animating, if we were closing, fire the close callback. */
-  const onRest = useCallback(() => {
-    if (closing.current) {
-      onClose();
-    }
-  }, [onClose]);
-
-  /** This spring is controlled via setSpring, which doesn't trigger re-render. */
-  const [springProps, setSpring] = useSpring(() => ({
-    // Initially transition from offscreen to on
-    from: { transform: `translateY(${window.innerHeight}px)` },
-    to: { transform: `translateY(0px)` },
-    config: spring,
-    immediate: reducedMotion,
-    onRest,
-  }));
+  const reducedMotion = Boolean(useReducedMotion());
+  const animationControls = useAnimation();
+  const dragControls = useDragControls();
 
   /**
-   * Closing the sheet sets closing to true and starts an animation to close. We only fire the
-   * outer callback when the animation is done.
+   * Triggering close starts the animation. The onClose prop is called by the callback
+   * passed to the onAnimationComplete motion prop
    */
-  const handleClose = useCallback(
-    (e?: React.MouseEvent, dragDismiss?: boolean) => {
+  const triggerClose = useCallback(
+    (e?: React.MouseEvent) => {
       if (disabled) {
         return;
       }
       e?.preventDefault();
-      closing.current = true;
       // Animate offscreen
-      setSpring.start({
-        immediate: reducedMotion && !dragDismiss,
-        to: { transform: `translateY(${height()}px)` },
-      });
+      animationControls.start('close');
     },
-    [setSpring, disabled, reducedMotion]
+    [disabled, animationControls]
   );
 
   // Handle global escape key
-  useGlobalEscapeKey(handleClose);
-
-  // This handles all drag interaction. The callback is called without re-render.
-  const bindDrag = useDrag(({ event, active, movement, velocity, last, cancel }) => {
-    event?.stopPropagation();
-
-    if (isEventFromFirefoxScrollbar(event as any)) {
-      cancel();
-    }
-
-    // If we haven't enabled dragging, cancel the gesture
-    if (!last && cancel && !dragging.current) {
-      cancel();
-    }
-
-    // How far down should we be positioned?
-    const yDelta = active ? Math.max(0, movement[1]) : 0;
-    // Set immediate (no animation) if we're in a gesture, so it follows your finger precisely
-    setSpring.start({ immediate: active, to: { transform: `translateY(${yDelta}px)` } });
-
-    // Detect if the gesture ended with a high velocity, or with the sheet more than
-    // dismissAmount percent of the way down - if so, consider it a close gesture.
-    if (
-      last &&
-      (movement[1] > (height() || 0) * dismissAmount ||
-        velocity[1] * Math.sign(movement[1]) > dismissVelocity)
-    ) {
-      handleClose(undefined, true);
-    }
-  });
+  useGlobalEscapeKey(triggerClose);
+
+  // We need to call the onClose callback when then close animation is complete so that
+  // the calling component can unmount the sheet
```

Comment:
I had the same though but figured I would keep the changes more in line with what we had for this first pass.
```suggestion
  // We need to call the onClose callback when then close animation is complete so that
  // the calling component can unmount the sheet
  // TODO (ryan/ben) move to using a container component and AnimatePresence
```